src/MaterialDesignThemes.Wpf/NumericUpDown.cs

```diff
@@ -256,4 +263,16 @@ protected override void OnPreviewMouseWheel(MouseWheelEventArgs e)
         }
         base.OnPreviewMouseWheel(e);
     }
+
+    private static void OnGotFocus(object sender, RoutedEventArgs e)
+    {
+        // When NumericUpDown gets focus move it to the TextBox
+        NumericUpDown numericUpDown = (NumericUpDown)sender;
+        if ((!e.Handled) && (numericUpDown._textBoxField != null))
+        {
+            numericUpDown._textBoxField.Focus();
+            numericUpDown._textBoxField.SelectAll();
```

Comment:
Though I typically like this SelectAll behavior on my apps I am hesitant to add it to the control by default. What about simply adding a new SelectAll method to this control so that apps can call it if they want that behavior?

Reply:
Yes, as I stated above it's debatable if `SelectAll` should be called inside of the event handler or not.
I guess with your suggestion we get the best of both worlds.